kube-client/src/api/remote_command.rs

```diff
@@ -202,71 +257,79 @@ where
     let (mut server_send, raw_server_recv) = stream.split();
     // Work with filtered messages to reduce noise.
     let mut server_recv = raw_server_recv.filter_map(filter_message).boxed();
-    let mut server_msg = server_recv.next();
-    let mut next_stdin = stdin_stream.next();
+    let have_terminal_size_rx = terminal_size_rx.is_some();
 
     loop {
-        match select(server_msg, next_stdin).await {
-            // from server
-            Left((Some(message), p_next_stdin)) => {
-                match message {
-                    Ok(Message::Stdout(bin)) => {
+        let terminal_size_next = async {
+            match terminal_size_rx.as_mut() {
+                Some(tmp) => Some(tmp.next().await),
+                None => None,
+            }
+        };
```

Comment:
here `terminal_size_next` is not awaited. Or maybe I'm misunderstanding something here.

Reply:
`tmp.next()` is awaited though. if we omitted that `.await` you'd have a future that the select could await in theory, and it would match the rest of the loop. not sure if it works out nicely with the current option wrapping though.

Reply:
oh, wait, ignore me, it's in an async block, so it's not actually awaited :facepalm: